common/transport/mqtt/src/main/java/org/thingsboard/server/transport/mqtt/session/AbstractGatewaySessionHandler.java

```diff
@@ -452,148 +462,165 @@ public TransportProtos.PostTelemetryMsg postTelemetryMsgCreated(TransportProtos.
 
     private void onDeviceClaimJson(int msgId, ByteBuf payload) throws AdaptorException {
         JsonElement json = JsonMqttAdaptor.validateJsonPayload(sessionId, payload);
-        if (json.isJsonObject()) {
-            JsonObject jsonObj = json.getAsJsonObject();
-            for (Map.Entry<String, JsonElement> deviceEntry : jsonObj.entrySet()) {
-                String deviceName = deviceEntry.getKey();
-                Futures.addCallback(checkDeviceConnected(deviceName),
-                        new FutureCallback<>() {
-                            @Override
-                            public void onSuccess(@Nullable T deviceCtx) {
-                                if (!deviceEntry.getValue().isJsonObject()) {
-                                    throw new JsonSyntaxException(CAN_T_PARSE_VALUE + json);
-                                }
-                                try {
-                                    DeviceId deviceId = deviceCtx.getDeviceId();
-                                    TransportProtos.ClaimDeviceMsg claimDeviceMsg = JsonConverter.convertToClaimDeviceProto(deviceId, deviceEntry.getValue());
-                                    processClaimDeviceMsg(deviceCtx, claimDeviceMsg, deviceName, msgId);
-                                } catch (Throwable e) {
-                                    log.warn("[{}][{}][{}] Failed to convert claim message: [{}]", gateway.getTenantId(), gateway.getDeviceId(), deviceName, deviceEntry.getValue(), e);
-                                }
-                            }
+        if (!json.isJsonObject()) {
+            throw new JsonSyntaxException(CAN_T_PARSE_VALUE + json);
+        }
+        JsonObject jsonObj = json.getAsJsonObject();
+        for (Map.Entry<String, JsonElement> deviceEntry : jsonObj.entrySet()) {
+            if (!deviceEntry.getValue().isJsonObject()) {
+                log.warn("{}[{}]", CAN_T_PARSE_VALUE, json);
+                continue;
+            }
 
-                            @Override
-                            public void onFailure(Throwable t) {
-                                log.debug("[{}][{}][{}] Failed to process device claiming command: [{}]", gateway.getTenantId(), gateway.getDeviceId(), sessionId, deviceName, t);
-                            }
-                        }, context.getExecutor());
+            String deviceName = deviceEntry.getKey();
+            T deviceCtx = devices.get(deviceEntry.getKey());
+
+            if (deviceCtx != null) {
+                processClaimDeviceMsg(deviceCtx, deviceEntry.getValue(), deviceName, msgId);
+            } else {
+                Futures.addCallback(checkDeviceConnected(deviceName), new FutureCallback<>() {
```

Comment:
checkDeviceConnected and `onDeviceConnect(deviceName, DEFAULT_DEVICE_TYPE)` are doing the same checks and wraps 
we can remove one of the method

the onDeviceConnect do all the job
![image](https://github.com/thingsboard/thingsboard/assets/79898499/4bbe981f-8f79-4a75-8dd8-f21e18387168)